dev-tools/validation/test_check_dev_demo_summary.py

```
@@ -0,0 +1,766 @@
+    def test_missing_helper_fails_closed_with_clear_assertion(self):
+        validator = self.validator
+        with tempfile.TemporaryDirectory() as directory:
+            root = Path(directory)
+            source = [
+                validator.WorkflowRunSource(
+                    "job",
+                    "step",
+                    'bash dev-tools/missing-summary.sh >> "$GITHUB_STEP_SUMMARY"',
+                )
+            ]
+            with self.assertRaisesRegex(
```

Ruff (0.16.0)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)